pkg/p2p/libp2p/connections_test.go

```diff
@@ -308,3 +278,41 @@ func TestConnectWithDisabledQUICAndWSTransports(t *testing.T) {
 	expectPeers(t, s2, overlay1)
 	expectPeersEventually(t, s1, overlay2)
 }
+
+func TestConnectRepeatHandshake(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	s1, overlay1, cleanup1 := newService(t, libp2p.Options{NetworkID: 1})
+	defer cleanup1()
+
+	s2, overlay2, cleanup2 := newService(t, libp2p.Options{NetworkID: 1})
```

Comment:
do we have a test that checks if two nodes on different `NetworkID`s can (or cannot) connect to each other?

Reply:
There is a test in handshake, I don't see the test in libp2p as well. Do you think we would need one?

Reply:
I think that the handshake test with different network ids covers this situation, as it is related to the handshake.